pilot/pkg/model/push_context.go

```diff
@@ -2262,6 +2263,16 @@ func (ps *PushContext) ServiceAccounts(hostname host.Name, namespace string, por
 	}]
 }
 
+func (ps *PushContext) SupportsTunnel(ip string) bool {
+	infos, _ := ps.ambientIndex.PodInformation(sets.New(types.NamespacedName{Name: ip}))
+	for _, p := range infos {
+		if p.Protocol == workloadapi.Protocol_HTTP {
```

Comment:
Q - I saw we no longer using the redirection annotation to detect if it is in ambient.  How would protocol http here help us detect if supports tunnel or not?

Reply:
I updated the PR description with a bunch of info.

tl;dr Protocol_HTTP is derived from annotation, we are just using the derived info instead of the source (which was a leaky abstraction)

Reply:
Thanks, very helpful!